### DISCUSSION TITLE
Updating from quarkus-rest-client to quarkus-rest-client-reactive-jackson

### DISCUSSION THREAD

**Discussion by Paul6552** [quarkusio/quarkus#33361] [Q&A]
Because I tried to update from version 2.16.7 to quarkus 3.0.3 and it failed great, i think i need to update some dependencies first. So I've now tackled `quarkus-rest-client`, changing this to `quarkus-rest-client-reactive-jackson`.

Although I have read the instructions [here](https://quarkus.io/guides/rest-client-reactive) carefully, I unfortunately get the error message:

```
Unsatisfied dependency for type at.tanzauswertung.portal.shared.RestClient.TurniersoftwareClient and qualifiers [@RestClient]
- java member: at.tanzauswertung.portal.meisterschaft.service.MeisterschaftService#turniersoftwareClient
- declared on CLASS bean [types=[java.lang.Object, at.tanzauswertung.portal.meisterschaft.service.MeisterschaftService], qualifiers=[@Default, @Any], target=at.tanzauswertung.portal.meisterschaft.service. ChampionshipDeleteService]
```

Unfortunately, after several hours I can't get it right. What am I doing wrong please?

My files:
**MeisterschaftService:**
```
@ApplicationScoped
@Transactional( Transactional.TxType.REQUIRED )
public class MeisterschaftService {

    @Inject
    MeisterschaftDao meisterschaftDao;
    @RestClient
    TurniersoftwareClient turniersoftwareClient; **//Here is the RestClient**
    @Inject
    SomethingDao awardMapper;
 ...
```

**My Interface with the Registered Rest Client**
```
import org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders;
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
import javax.ws.rs.Path;
import static at.tanzauswertung.api.rest.constants.PortalRestPaths.TURNIERSOFTWARE_ENDPOINT;

@Path( TURNIERSOFTWARE_ENDPOINT )
@RegisterRestClient( configKey = "turniersoftware-api" )
@RegisterClientHeaders
public interface TurniersoftwareClient extends MusicApi, MesterschaftTurniersoftwareApi, LetzteTanzIDApi {
}
```

**Pom.xml:**
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd"
         xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <modelVersion>4.0.0</modelVersion>

    <parent>
        <groupId>at.tanzauswertung.portal</groupId>
        <artifactId>portal-parent</artifactId>
        <version>1.0-SNAPSHOT</version>
    </parent>

    <groupId>at.tanzauswertung.portal-backend</groupId>
    <artifactId>portal-backend</artifactId>

    <properties>
        <webapp.dependency.phase>prepare-package</webapp.dependency.phase>
        <surefire-plugin.version>3.0.0-M5</surefire-plugin.version>
        <lombok.version>1.18.26</lombok.version>
        <keycloak.version>20.0.3</keycloak.version>
        <testcontainers.version>1.17.6</testcontainers.version>
        <quarkus.version>2.16.7.Final</quarkus.version>
        <version.apache-poi>5.2.2</version.apache-poi>
        <maven.compiler.source>17</maven.compiler.source>
        <maven.compiler.target>17</maven.compiler.target>
        <compiler-plugin.version>3.8.1</compiler-plugin.version>
        <docker-plugin.version>0.28.0</docker-plugin.version>
    </properties>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>io.quarkus</groupId>
                <artifactId>quarkus-universe-bom</artifactId>
                <version>${quarkus.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <dependencies>
        <dependency>
            <groupId>at.tanzauswertung.api</groupId>
            <artifactId>pea-portal-api</artifactId>
            <version>1.0-SNAPSHOT</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.datatype</groupId>
            <artifactId>jackson-datatype-jsr310</artifactId>
            <version>2.14.2</version>
        </dependency>
        <dependency>
            <groupId>net.sf.jasperreports</groupId>
            <artifactId>jasperreports</artifactId>
            <version>6.20.1</version>
            <exclusions>
                <exclusion>
                    <groupId>com.fasterxml.jackson.core</groupId>
                    <artifactId>jackson-core</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.fasterxml.jackson.core</groupId>
                    <artifactId>jackson-databind</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.fasterxml.jackson.core</groupId>
                    <artifactId>jackson-annotations</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.apache.lucene</groupId>
                    <artifactId>lucene-core</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.fasterxml.jackson.core</groupId>
                    <artifactId>lucene-analyzers-common</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.apache.lucene</groupId>
                    <artifactId>lucene-queryparser</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.jfree</groupId>
                    <artifactId>jcommon</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.jfree</groupId>
                    <artifactId>jfreechart</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.eclipse.jdt.core.compiler</groupId>
                    <artifactId>ecj</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.codehaus.castor</groupId>
                    <artifactId>castor-xml</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.olap4j</groupId>
                    <artifactId>olap4j</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.google.zxing</groupId>
                    <artifactId>core</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.ibm.icu</groupId>
                    <artifactId>icu4j</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>org.apache.poi</groupId>
            <artifactId>poi</artifactId>
            <version>${version.apache-poi}</version>
        </dependency>
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-lang3</artifactId>
            <version>3.12.0</version>
        </dependency>
        <dependency>
            <groupId>org.apache.poi</groupId>
            <artifactId>poi-ooxml</artifactId>
            <version>${version.apache-poi}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-keycloak-admin-client-reactive</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-arc</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-mailer</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-narayana-jta</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-hibernate-orm</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-hibernate-orm-panache</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-scheduler</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>org.jboss.spec.javax.ws.rs</groupId>
            <artifactId>jboss-jaxrs-api_2.1_spec</artifactId>
            <version>2.0.2.Final</version>
        </dependency>
        <dependency>
            <groupId>org.hibernate</groupId>
            <artifactId>hibernate-jpamodelgen</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-agroal</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-rest-client-reactive-jackson</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
        </dependency>
        <dependency>
            <groupId>org.apache.httpcomponents</groupId>
            <artifactId>httpmime</artifactId> **//I know, I have to change this as well**
            <version>4.5.14</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-jdbc-postgresql</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-undertow</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-keycloak-authorization</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-oidc</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>rest-assured</artifactId>
            <version>5.3.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-junit5</artifactId>
            <version>${quarkus.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-test-h2</artifactId>
            <version>${quarkus.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.testcontainers</groupId>
            <artifactId>junit-jupiter</artifactId>
            <version>${testcontainers.version}</version>
        </dependency>
        <dependency>
            <groupId>org.testcontainers</groupId>
            <artifactId>postgresql</artifactId>
            <version>${testcontainers.version}</version>
        </dependency>
        <dependency>
            <groupId>com.github.dasniko</groupId>
            <artifactId>testcontainers-keycloak</artifactId>
            <version>2.4.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-junit5-mockito</artifactId>
            <version>${quarkus.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-flyway</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>${lombok.version}</version>
            <scope>provided</scope>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>${surefire-plugin.version}</version>
                <configuration>
                    <skipTests>true</skipTests>
                    <systemProperties>
                        <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
                    </systemProperties>
                </configuration>
            </plugin>
            <plugin>
                <groupId>io.quarkus</groupId>
                <artifactId>quarkus-maven-plugin</artifactId>
                <version>${quarkus.version}</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>build</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-resources-plugin</artifactId>
                <version>3.2.0</version>
                <executions>
                    <execution>
                        <id>copy-resources</id>
                        <phase>validate</phase>
                        <goals>
                            <goal>copy-resources</goal>
                        </goals>
                        <configuration>
                            <outputDirectory>${project.build.directory}/classes/META-INF/resources/portal
                            </outputDirectory>
                            <resources>
                                <resource>
                                    <directory>${project.parent.basedir}/portal-webapp/build/</directory>
                                </resource>
                            </resources>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>

    <profiles>
        <profile>
            <id>skipFrontendBuild</id>
            <properties>
                <surefire-plugin.version>${surefire-plugin.version}</surefire-plugin.version>
                <webapp.dependency.phase>none</webapp.dependency.phase>
            </properties>
        </profile>
    </profiles>

</project>

**Application.properties:**
```
#Rest-api settings
#turniersoftware-api/mp-rest/url=http://localhost:8080/ **//Those were the old settings.**
#turniersoftware-api/mp-rest/scope=javax.inject.Singleton **//Those were the old settings.**
quarkus.rest-client.turniersoftware-api.url=http://localhost:8080/ **//These are the new settings**
quarkus.rest-client.turniersoftware-api.scope=jakarta.inject.Singleton**//These are the new settings**
quarkus.rest-client.turniersoftware-api.verify-host=false**//These are the new settings**
org.eclipse.microprofile.rest.client.propagateHeaders=Authorization
```

**Comment by dyutiman** [comment]
How did this get resolved?

**Reply by Paul6552** [reply]
Jandex index was missing. With version 3.0.0 this is mandatory when you use more modules/repositories

https://quarkus.io/guides/cdi-reference
